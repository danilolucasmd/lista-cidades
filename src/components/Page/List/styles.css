:local(.component) table {
  font-family: arial, sans-serif;
  border-collapse: collapse; }

:local(.component) td, :local(.component) th {
  border: 1px solid #000000;
  text-align: left;
  padding: 8px; }

:local(.component) tr:nth-child(odd) {
  background-color: #dddddd; }
