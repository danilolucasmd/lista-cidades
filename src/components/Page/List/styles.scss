:local(.component) {
  // width: 500px;

  table {
    font-family: arial, sans-serif;
    border-collapse: collapse;
  }

  td, th {
      border: 1px solid #000000;
      text-align: left;
      padding: 8px;
  }

  tr:nth-child(odd) {
      background-color: #dddddd;
  }
}