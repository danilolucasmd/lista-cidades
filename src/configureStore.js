import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import app from './reducers';

const configureStore = () =>
  createStore(
    app,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
    applyMiddleware(thunk)
  );

export default configureStore;
