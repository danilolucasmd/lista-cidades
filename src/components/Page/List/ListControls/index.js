import React from 'react';

import styles from './styles.css';

const ListControls = ({ nextPage, prevPage, curPage, lastPage }) => (
  <div className={styles.component}>
    <button onClick={() => curPage > 1 && prevPage()}>
      Anterior
    </button>
    <label>{curPage}</label>
    <button onClick={() => curPage < lastPage && nextPage()}>
      Pr√≥ximo
    </button>
  </div>
);

export default ListControls;