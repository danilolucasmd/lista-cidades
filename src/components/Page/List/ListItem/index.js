import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchCityPoints } from 'actionCreators/cities';

import styles from './styles.css';

const ListItem = ({ data, fetchCityPoints }) => (
  <tr className={styles.component}>
    <td>{data.Estado}</td>
    <td>{data.Nome}</td>
    <td>
      <a href='' onClick={e => {
        e.preventDefault();

        fetchCityPoints(data);
      }}>
        Ver Pontuação
      </a>
    </td>
  </tr>
);

ListItem.propTypes = {
  data: PropTypes.object.isRequired,
};

export default connect(undefined, { fetchCityPoints })(ListItem);