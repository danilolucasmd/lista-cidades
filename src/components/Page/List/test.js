import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './index';

import configureStore from 'configureStore';

const store = configureStore();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<List store={store} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(
    <List store={store} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});