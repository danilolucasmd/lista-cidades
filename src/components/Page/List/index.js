import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchCities } from 'actionCreators/cities';
import { getVisibleCities } from 'reducers';
import ListItem from './ListItem';
import ListControls from './ListControls';

import styles from './styles.css';

class List extends Component {
  PAGE_SIZE = 10;

  constructor(props) {
    super(props);

    this.state = {
      curPage: 1,
      firstItem: 0,
      lastItem: this.PAGE_SIZE,
    };
  }

  componentDidMount() {
    this.props.fetchCities();
  }

  componentWillReceiveProps() {
    this.setState({
      curPage: 1,
      firstItem: 0,
      lastItem: this.PAGE_SIZE,
    });
  }

  handleNextPage = () => {
    this.setState({
      firstItem: this.state.firstItem + this.PAGE_SIZE,
      lastItem: this.state.lastItem + this.PAGE_SIZE,
      curPage: this.state.curPage + 1,
    })
  }

  handlePreviousPage = () => {
    this.setState({
      firstItem: this.state.firstItem - this.PAGE_SIZE,
      lastItem: this.state.lastItem - this.PAGE_SIZE,
      curPage: this.state.curPage - 1,
    })
  }

  render() {
    return (
      <div className={styles.component}>
        <table>
          <thead>
            <tr>
              <th>Estado</th>
              <th>Cidade</th>
              <th></th>
            </tr>
          </thead>
          <tbody>
            {this.props.data.slice(this.state.firstItem, this.state.lastItem).map((item, index) => {
              return (
                <ListItem
                  key={index}
                  data={item}
                />
              );
            })}
          </tbody>
        </table>

        <ListControls
          nextPage={this.handleNextPage}
          prevPage={this.handlePreviousPage}
          curPage={this.state.curPage}
          lastPage={this.props.data.length / this.PAGE_SIZE}
        />
      </div>
    );
  }
}

List.propTypes = {
  data: PropTypes.array.isRequired,
  fetchCities: PropTypes.func,
};

const mapStateToProps = state => ({
  data: state.filters.city || state.filters.state ? getVisibleCities(state.cities, state.filters) : state.cities,
});

export default connect(mapStateToProps, { fetchCities })(List);