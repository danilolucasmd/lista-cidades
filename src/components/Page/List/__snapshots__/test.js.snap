// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className={undefined}
>
  <table>
    <thead>
      <tr>
        <th>
          Estado
        </th>
        <th>
          Cidade
        </th>
        <th />
      </tr>
    </thead>
    <tbody />
  </table>
  <div
    className={undefined}
  >
    <button
      onClick={[Function]}
    >
      Anterior
    </button>
    <label>
      1
    </label>
    <button
      onClick={[Function]}
    >
      Pr√≥ximo
    </button>
  </div>
</div>
`;
